plugin.rb

```diff
@@ -375,6 +380,20 @@ def before_head_close_meta(controller)
     end
   end
 
+  if respond_to?(:register_modifier)
+    register_modifier(:search_rank_sort_priorities) do |priorities, search|
+      priorities << [<<~SQL, 1.1] if SiteSetting.prioritize_solved_topics_in_search
+            EXISTS
+              (
+                SELECT 1 FROM topic_custom_fields f
+                WHERE topics.id = f.topic_id AND
+                  f.name = '#{::DiscourseSolved::ACCEPTED_ANSWER_POST_ID_CUSTOM_FIELD}'
+              )
+          SQL
+      priorities
```

Comment:
```suggestion
      if SiteSetting.prioritize_solved_topics_in_search 
        condition = <<~SQL
          EXISTS
            (
              SELECT 1 FROM topic_custom_fields f
              WHERE topics.id = f.topic_id 
              AND f.name = '#{::DiscourseSolved::ACCEPTED_ANSWER_POST_ID_CUSTOM_FIELD}'
            )
        SQL

        priorities.push([condition, 1.1])
      end
```

Minor suggestion because I find the existing format hard to read.

Reply:
I don't think that works... we need to return priorities unconditionally. unfortunately the auto formatting here kind of bit us. 

The change means that if the setting is disabled the plugin blows away all the priorities cause it returns false. Modifiers are tricky things and need to be used very carefully, you need to be mindful always to return the right data from a modifier.

Reply:
```suggestion
      if SiteSetting.prioritize_solved_topics_in_search 
        condition = <<~SQL
          EXISTS
            (
              SELECT 1 FROM topic_custom_fields f
              WHERE topics.id = f.topic_id 
              AND f.name = '#{::DiscourseSolved::ACCEPTED_ANSWER_POST_ID_CUSTOM_FIELD}'
            )
        SQL

        priorities.push([condition, 1.1])
      else
        priorities
      end
```

O good catch. This should work?

Reply:
yes that looks good to me!